src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
+          }));
+        }
+        authStore.update((state) => ({
+          ...state,
+          authTokenMetadata: {
+            ...authStore.get().authTokenMetadata,
+            token: user.access_token,
+            createdAt: authTokenCreatedAt,
+            expiresAt: authTokenExpiresAt,
+            id: authTokenId,
+            // TODO: Change this to only update if the auth token is different than the previous token
+            totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+          },
+        }));
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+          authProvider: (user.profile as B2cIdTokenClaims).idp,
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+          authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+          authTokenId,
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+          jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+        });
+      },
+    ],
+    [
+      'expired',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ],
+    [
+      'error',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+          // we could potentially log the reason, but I don't know if that data is safe to log
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ]
+  );
+
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null | AuthTokenErrorState = await (popUp
+    ? // returns Promise<User | null>, attempts to use the refresh token to get a new authToken
+      authInstance.signinPopup(args)
+    : authInstance.signinSilent(args)
+  ).catch((e) => {
+    const errorMsg = 'An unexpected exception occurred while attempting to load auth credentials.';
```

Comment:
Is this error message user facing?

If so, I'd suggest something with less technical detail/jargon.

```suggestion
    const errorMsg = 'Unable to sign in.';
```

Re: reusing the error message for the ErrorState and the console message... generally, user facing and developer facing error messages serve very different purposes. I'd suggest tailoring each to the intended audience.

Reply:
Do we consider the console.error messages to be user facing? Or are you saying that the errorMsg should be what we propagate back to the user? I am not sure I follow

Reply:
I would not consider a `console.error` message to be user-facing. A user could see it, but if they're looking at the console, they likely either have technical knowledge themselves or may be following instructions from Support to look for more detailed information.

But for messages shown in error notifications, etc. let's avoid the technical details and jargon (exception, auth credentials, etc).
https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4

Reply:
Currently, it does not send this error to the user, but we added specific properties to ErrorState which can store internal error message as well as user error messages.